### ISSUE TITLE
False positive for NoUnusedImports

### ISSUE THREAD

**Issue by danikula** [detekt/detekt#3265]
## Expected Behavior
Detekt should not report `NoUnusedImports` for used import.

## Steps to Reproduce
Bar.kt
```
package detekt.bar

fun `is`() = Unit
```

Foo.kt
```
package detekt

import detekt.bar.`is` as is2

fun foo() = is2()
```

Detekt reports issue `NoUnusedImports` in line ```import detekt.bar.`is` as is2```

## Your Environment
* Version of detekt used: 1.15.0-RC1
* Operating System and version: Mac OS 10.15.7

**Comment by cortinico** [comment]
The `NoUnusedImports` rule is a rule that we mirror from KtLint: https://github.com/pinterest/ktlint/blob/master/ktlint-ruleset-standard/src/main/kotlin/com/pinterest/ktlint/ruleset/standard/NoUnusedImportsRule.kt

You should open a similar issue on https://github.com/pinterest/ktlint (perhaps they already have a similar issue resolved https://github.com/pinterest/ktlint/issues/910).

Closing as this is unrelated to us.
